File: Algorithms/Stack/496. Next Greater Element I.py
class Solution(object):
	def nextGreaterElement(self, findNums, nums):
		hashmap = dict()
		stack = list()
		if(nums):
			stack.append(nums[0])
		for num in nums[1:]:
			if(num <= stack[-1]):
				stack.append(num)
			else:
				while(stack and stack[-1] < num):
					hashmap[stack.pop()] = num
				stack.append(num)
		ret = list()
		for num in findNums:
			if(num in hashmap):
				ret.append(hashmap[num])
			else:
				ret.append(-1)
		return ret

File: Algorithms/Hash Table/496. Next Greater Element I.java
class Solution {
    public int[] nextGreaterElement(int[] nums1, int[] nums2) {
        int[] res = new int[nums1.length];
        HashMap<Integer, Integer> hash = new HashMap<>();
        Stack<Integer> stack = new Stack<>();
        for(int num : nums2) {
            while(!stack.empty() && num > stack.peek()) {
                hash.put(stack.pop(), num);
            }
            stack.push(num);
        }
        
        for(int inx = 0; inx < nums1.length; ++inx) {
            res[inx] = hash.getOrDefault(nums1[inx], -1);
        }
        return res;
    }
}
